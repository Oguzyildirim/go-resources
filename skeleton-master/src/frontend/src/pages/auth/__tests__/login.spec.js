import { mount } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import Login from '../login'

describe('login.vue', () => {
  let actions
  let store

  beforeEach(() => {
    actions = {
      'page/title': jest.fn(),
      'loading/start': jest.fn(),
      'loading/finish': jest.fn(),
      'auth/login': jest.fn()
    }
    store = new Vuex.Store({
      actions,
      state: {
        loading: { status: false }
      }
    })
  })

  it('should match snapshot', () => {
    const w = mount(Login, {
      store,
      stubs: ['router-link', 'router-view']
    })

    expect(w.html()).toMatchSnapshot()
    w.destroy()
  })

  it('can process login', async () => {
    const router = new VueRouter({})
    const w = mount(Login, {
      store,
      router,
      stubs: ['router-link', 'router-view']
    })

    w.find('[data-username]').setValue('user1')
    w.find('[data-password]').setValue('pass1')
    w.find('form').trigger('submit.prevent')
    await w.vm.$nextTick()
    await w.vm.$nextTick()

    expect(w.vm.$route.path).toEqual('/dashboard/home/')
    expect(w.find('.v-alert__content > div').text())
      .toBe('Successfully logged in!')
    w.destroy()
  })
})
