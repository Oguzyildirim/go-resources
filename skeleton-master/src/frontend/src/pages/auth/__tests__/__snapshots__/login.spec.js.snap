// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login.vue should match snapshot 1`] = `
<section class="container container--fluid">
  <div class="row justify-center">
    <div class="v-card--material pa-3 pa-6 v-card v-sheet theme--light v-card--material--has-heading" style="min-width: 20%;" min-width="20%">
      <div class="d-flex grow flex-wrap">
        <div class="text-start v-card--material__heading mb-n6 v-sheet theme--dark elevation-6 info pa-7" style="width: 100%;">
          <div class="display-2 font-weight-light">
            Login
          </div>
          <div class="subtitle-1 font-weight-light">
            Fill in your username and password
          </div>
          <!---->
          <!---->
        </div>
        <!---->
      </div> <span> <!----> <!----></span>
      <form novalidate="novalidate" method="post" class="v-form pt-8">
        <div class="container py-0">
          <div class="row">
            <div class="v-input theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span>​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-9" class="v-label theme--light" style="right: 0px; position: absolute;">Username*</label><input data-username="" id="input-9" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="row">
            <div class="v-input theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span>​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-12" class="v-label theme--light" style="right: 0px; position: absolute;">Password*</label><input data-password="" id="input-12" type="password"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="row">
            <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">$checkboxOff</i><input aria-checked="false" id="input-15" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-15" class="v-label theme--light" style="left: 0px; position: relative;">Remember me</label>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="row">
            <router-link-stub to="/auth/forgot-password/" tag="a" activeclass="v-btn--active" exactactiveclass="v-btn--active" ariacurrentvalue="page" event="click" class="v-btn v-btn--flat v-btn--router v-btn--text theme--light v-size--small blue--text text--darken-1"><span class="v-btn__content">
                Forgot Password?
              </span></router-link-stub>
          </div>
          <div class="spacer"></div>
          <div class="text-right col"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default blue--text text--darken-1"><span class="v-btn__content">
                Reset
              </span></button> <button type="submit" class="mr-0 v-btn v-btn--contained theme--light v-size--default info"><span class="v-btn__content">
                Login
              </span></button></div>
        </div>
      </form>
      <div class="divider my-5"><span>Or use social networks</span></div>
      <div class="row align-center justify-center"><button type="button" class="v-btn v-btn--contained v-btn--fab v-btn--round theme--light v-size--x-small white"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-github theme--dark black--text" style="font-size: 34px;"></i></span></button> <button type="button" class="v-btn v-btn--contained v-btn--fab v-btn--round theme--light v-size--x-small white"><span class="v-btn__content"><svg width="32px" height="32px" viewBox="0 0 533.5 544.3" xmlns="http://www.w3.org/2000/svg"><path d="M533.5 278.4c0-18.5-1.5-37.1-4.7-55.3H272.1v104.8h147c-6.1 33.8-25.7 63.7-54.4 82.7v68h87.7c51.5-47.4 81.1-117.4 81.1-200.2z" fill="#4285f4"></path><path d="M272.1 544.3c73.4 0 135.3-24.1 180.4-65.7l-87.7-68c-24.4 16.6-55.9 26-92.6 26-71 0-131.2-47.9-152.8-112.3H28.9v70.1c46.2 91.9 140.3 149.9 243.2 149.9z" fill="#34a853"></path><path d="M119.3 324.3c-11.4-33.8-11.4-70.4 0-104.2V150H28.9c-38.6 76.9-38.6 167.5 0 244.4l90.4-70.1z" fill="#fbbc04"></path><path d="M272.1 107.7c38.8-.6 76.3 14 104.4 40.8l77.7-77.7C405 24.6 339.7-.8 272.1 0 169.2 0 75.1 58 28.9 150l90.4 70.1c21.5-64.5 81.8-112.4 152.8-112.4z" fill="#ea4335"></path></svg></span></button> <button type="button" class="v-btn v-btn--contained v-btn--fab v-btn--round theme--light v-size--x-small white"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-facebook theme--dark" style="font-size: 34px; color: rgb(59, 88, 153); caret-color: #3b5899;"></i></span></button> <button type="button" class="v-btn v-btn--contained v-btn--fab v-btn--round theme--light v-size--x-small white"><span class="v-btn__content"><svg width="28px" height="28px" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="#1b6499"><path d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-2 16h-2v-6h2v6zm-1-6.891c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zm8 6.891h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0v2.861h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548v3.359z"></path></svg></span></button></div>
      <!---->
    </div>
  </div>
</section>
`;
